src/common/io_exerciser/IoOp.cc

```diff
@@ -0,0 +1,188 @@
+#include "IoOp.h"
+
+using IoOp = ceph::io_exerciser::IoOp;
+
+IoOp::IoOp( OpType op,
+            uint64_t offset1, uint64_t length1,
+            uint64_t offset2, uint64_t length2,
+            uint64_t offset3, uint64_t length3) :
+  op(op),
+  offset1(offset1), length1(length1),
+  offset2(offset2), length2(length2),
+  offset3(offset3), length3(length3)
+{
+  
+}
+
+std::string IoOp::value_to_string(uint64_t v) const
+{
+  if (v < 1024 || (v % 1024) != 0) {
```

Comment:
Note that if performance was an issue (which I expect it is not here): to_string() is usually much slower than the alternatives (e.g. to_chars).

Reply:
I have not updated this and will take the performance hit for now. That is useful to know for future reference though when I am looking at code with higher performance requirements though, thank you.